### PR TITLE
removes policy action, adds bucketaccessinfo

### PR DESCRIPTION
**Pull request by krishchow**
as above, this PR removes the policyActionConfig map in the API, and it adds the new bucketAccesssInfo struct which is part of the COSI downwards API.

### CODE REVIEW

File: apis/objectstorage.k8s.io/v1alpha1/access_types.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+// +kubebuilder:resource:scope=Namespaced
+// +kubebuilder:storageversion
+
+type BucketAccessInfo struct {
+	metav1.TypeMeta `json:",inline"`
+
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	S3 	  *S3BucketAccessInfo    `json:"s3,omitempty"`
```
Comment (wlan0) [inline apis/objectstorage.k8s.io/v1alpha1/access_types.go:16]: Formatting

**Comment by wlan0** [comment]
Does this needs to be coupled with corresponding changes to other COSI repos?

**Comment by krishchow** [comment]
Yes, that's correct! There will be another PR shortly for the Spec, followed by later changes for the other components.

File: apis/objectstorage.k8s.io/v1alpha1/access_types.go [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+}
+
+type S3BucketAccessInfo struct {
+	Endpoint         string             `json:"endpoint,omitempty"`
+	BucketName       string             `json:"bucketName,omitempty"`
+	Region           string             `json:"region,omitempty"`
+	Credentials      string             `json:"credentials,omitempty"`
+	Certificates     string             `json:"certificates,omitempty"`
+	SignatureVersion S3SignatureVersion `json:"signatureVersion,omitempty"`
+}
```
Comment (xing-yang) [inline apis/objectstorage.k8s.io/v1alpha1/access_types.go:28]: I thought we are going to remove "SignatureVersion" and "Certificates"?

**Pull request closed without merging**